feat(dynamic-sampling): organization settings (#11791)
getsentry/sentry-docs: docs/organization/dynamic-sampling/index.mdx

**Project Preview**

The [project preview](https://sentry.io/orgredirect/organizations/:orgslug/settings/dynamic-sampling/) table on the Dynamic Sampling settings page shows you how your project might be affected by the sample rate you've chosen. The preview shows the dynamically allocated rates for your projects based on either the last 24 hours or the last 30 days of traffic.

![The dynamic sampling mode and target sample rate setting.](./img/project-sample-rate-preview.png)

Because project rates continuously change, we recommend that you choose the 24-hour preview to get a more accurate view of your current sample rates for real-time analysis. We recommend the 30-day preview for seeing average sample rates in your billing cycle for long-term planning.

<Alert>
  If you deactivate advanced mode, the recalibration process will
  start immediatly and the per-project sample rates will be reset to the
  calculated values. There's no way to restore the manually configured project
  sample rates once you've deactivated advanced mode.

> </Alert>

Reviewer: I would move this into the "Advanced Mode" section below a paragraph that describes that you can deactivate advanced mode and go back to configuring the organization-wide target rate.

### Manual mode

In the "manual" mode, you can configure a fixed sample rate for each project in your organization. This value will specify how much of the incoming traffic should be stored for the respective project. The sample rate for each project can be set between 0% and 100% and it defaults to 100% and are not adjusted by Dynamic Sampling.

When the sample rates of each project is edited, an estimated overall organization sample rate is calculated as a weighted average across all projects. This value is always an estimation and changes according to volume within projects.

> If you want to change the sample rates for all projects at once, you an edit the overall organition sample rate. This will in turn adjust the sample rates of all projects to achieve the estimate rate. The maximum value is 100% and other projects will be adjusted accordingly to achieve the estimate rate.

Reviewer: ```suggestion
If you want to change the sample rates for all projects at once, you an edit the overall organization sample rate. This will proportionally adjust the sample rates of all projects to achieve the estimate rate. The maximum value is 100% and other projects will be adjusted accordingly to achieve the estimate rate. You can make final changes to the updated sample rates before saving all projects at once.
```

In order to apply the changes, you need to click the "Apply changes" button, otherwise, the changes will not be saved.

> As with automatic mode [Dynamic Sampling Priorities](#dynamic-sampling-priorities) apply within the project, which will boost certain spans and deprioritize others.

Reviewer: ```suggestion
Like in automatic mode, [Dynamic Sampling Priorities](#dynamic-sampling-priorities) apply within each project, which boosts certain spans and deprioritizes others.
```

By analyzing incoming traffic patterns, Dynamic Sampling is able to tailor its decisions to both the actual traffic volume as well as the content of accepted spans. For more details, check out the [Dynamic Sampling Priorities](#dynamic-sampling-priorities) section.

<Alert>
  Dynamic Sampling is available for customers on our latest Enterprise plans,
  starting June, 2024.
</Alert>

## Prerequisites

- [Admin-level permissions](/organization/membership/).
- Latest version of one of the below SDKs:
  - Python: 1.7.2 or later
  - JavaScript: 7.6.0 or later
  - Apple: 7.23.0 or later
  - Android: 6.5.0 or later
  - React Native: 4.3.0 or later
  - Dart and Flutter: 6.11.0 or later
  - PHP: 3.9.0 or later
  - Laravel: 3.0.0 or later
  - Symfony: 4.4.0 or later
  - Ruby: 5.5.0 or later
  - Java: 6.5.0 or later
  - .NET: 3.22.0 or later
  - Go: 0.16.0 or later

## Configuration

In this section you'll learn how to use Dynamic Sampling in your organization.

![The dynamic sampling mode and target sample rate setting.](./img/mode-and-target-sample-rate.png)

### Modes

> If you prefer not to set sample rates manually, Sentry can do it for you. You just have to set a global sample rate for your organization and Sentry will then automatically balance the sample rates across projects, giving priority to factors like low-volume endpoints and new releases. If you want more fine-grained control, you can customize data retention by setting individual sample rates for specific projects.

Reviewer 2: Here I miss an introductory sentence, something like "we offer two modes: simple mode and advanced mode".

---
title: Dynamic Sampling
sidebar_order: 30
description: "Learn how to make your billing quota go farther with Sentry's Dynamic Sampling."
---

## Overview

High-volume projects that generate many events can quickly consume a large portion of your billing quota, leaving less room for other projects and decreasing visibility across them.

> With high-volume projects, you'll often see many similar events, which can be redundant and less helpful for debugging. But with Dynamic Sampling, you can maximize your quota by adjusting sample rates based on each project’s event volume. The sample rate is a percentage from 0% (no events passed) to 100% (all events passed), that can be tailored to each project’s needs.

Reviewer: This might be misleading. Without going into the details, there's always going to be a quota for how much is sent to Sentry by the SDKs. Dynamic Sampling has no effect on that, so you cannot just "maximize your quota". Sent spans is always going to be the smaller price component, so it makes sense to move more of the sampling to Sentry. The benefit of that is that Dynamic Sampling is smarter and adapts to changes, so you generally will get higher quality samples than with the random sampling in SDKs. We can further call out some examples from the priorities, like automatically boosting latest releases or boosting small endpoints.

> When you use Dynamic Sampling, projects in your organization are sampled based on their recent event volumes. High-volume projects are sampled at lower rates while lower-volume projects are sampled at higher rates or not sampled at all, based on your recent event data. This makes it possible for each project to retain a meaningful set of key events, improving visibility into all projects regardless of their size.

Reviewer: - For someone not too familiar with the matter, does "event volume" make sense in this context? For me knowing the internals this sounds wrong (we don't classify spans as events) but maybe for a user this is fine. 
- "not sampled" means that everything is dropped. I think you mean to say "sampled at higher rates up to 100%". Also, in this sentence, we don't sample based on event data, but based on event volume.

![The dynamic sampling mode and target sample rate setting.](./img/mode-and-target-sample-rate.png)

## Modes

You can choose between two Dynamic Sampling modes for your organization: manual or automatic.

    * **`automatic`** (default): In automatic mode, the sample rate for each project is set automatically based on relative project volume, so events fit within your organization’s billing quota. This boosts visibility into lower-volume projects. To account for any changes in event volume, sample rates are recalibrated on a regular basis. Automatic mode overrides any manually-set project sample rates. 
    
    * **`manual`**: If you want to set the exact sample rate for each project in your organization yourself, choose manual mode. Note, that in this mode, there is no automatic recalibration.

[]: # (TODO: Rename AM3 with the actual user-facing name.)

<Note>

> Please note that these Dynamic Sampling modes are only available for customers on the AM3 platform.

Reviewer: Should we call out that on AM2 there's also DS, the sample rate is chosen by the system based on transaction volume (1M+), and you still have controls for the sampling priorities?

</Note>

## Differences to Sampling in the Sentry SDKs

Dynamic Sampling is different from the sampling that you can configure in the Sentry SDKs. The SDK sampling rate is applied before events are sent to Sentry, while Dynamic Sampling is applied after events are received by Sentry. This means that Dynamic Sampling can be used to adjust the sample rate for events that have already been sent to Sentry and only take into account the volume of events that have been received by Sentry.

> Also in contrast to the Dynamic Sampling, the SDK sample rate is never dynamically adjusted.

Reviewer: Strictly speaking, "never" isn't true :) `traces_sampler` allows customers to write their own code for dynamically adjusting sample rates. I think the true differences can rather be summarized as:
- Sampling in SDKs affects how much is captured and sent to Sentry. For high throughput scenarios, this may be necessary to reduce instrumentation overheads.
- To change sample rates in SDKs, one has to update the `traces_sample_rate` or `traces_sampler` callback, which usually requires a new deployment or release.
- With dynamic sampling, one can change sample rates without touching the application or SDK. This affects the amount of data _stored_ at Sentry, however, it will not have an effect on how much is sent from the SDKs.
- Dynamic sampling continuously monitors incoming traffic patterns and adjusts fine-grained sample rates within the projects to prioritize more important data. See the sampling priorities for examples.

## Dynamic Sampling Priorities

<Note>

> Dynamic Sampling priorities are only available on the latest version of our plans, starting from 1M reserved transactions. Anyone in your org with owner or billing-level permissions can update your plan by going to Settings > Subscription, and clicking the "Manage Subscription" button.

Reviewer: This will need to be updated. DS priorities are available on AM2 _and_ AM3 (right now this is still correct, as AM2 is latest).
